src/Calculator/SettingsPage.xaml.cpp

```diff
@@ -0,0 +1,177 @@
+using namespace Windows::UI::Xaml::Data;
+using namespace winrt::Windows::Storage;
+
+#ifndef BUILD_YEAR
+#define BUILD_YEAR 2021
+#endif
+
+auto resourceLoader = AppResourceProvider::GetInstance();
+auto themevalue = Windows::Storage::ApplicationData::Current -> LocalSettings -> Values -> Lookup(L"themeSetting");
+String ^ colorSetting = safe_cast<String ^>(themevalue);
+
+SettingsPage::SettingsPage()
+{
+    auto locService = LocalizationService::GetInstance();
+
+	InitializeComponent();
+
+    Language = locService->GetLanguage();
+
+    this->SetVersionString();
+    this->SetCopyrightString();
+
+    
+    if (themevalue != nullptr)
+    {
+        if (colorSetting == L"Dark")
+        {
+            SettingsDarkTheme->IsChecked = true;
+        }
+        else if (colorSetting == L"Light")
+        {
+            SettingsLightTheme->IsChecked = true;
+        }
+        else if (colorSetting == L"System")
+        {
+            SettingsSystemTheme->IsChecked = true;
+        }
+    }
+    else
+    {
+        SettingsSystemTheme->IsChecked = true;
+    }
+
+    InitializeContributeTextBlock();
+}
+
+void SettingsPage::OnNavigatedTo(Windows::UI::Xaml::Navigation::NavigationEventArgs ^ e)
+{
+    MainPage ^ mainPage = (MainPage ^) e->Parameter;
+    if (mainPage != nullptr)
+    {
+        MainPageProperty = mainPage;
+    }
+}
+
+void SettingsPage::InitializeContributeTextBlock()
+{
+    std::wstring contributeHyperlinkText  = resourceLoader->GetResourceString(L"SettingsContribute")->Data();
+
+    // The resource string has '%HL%' wrapped around 'GitHub'
+    // Break the string and assign pieces appropriately.
+    static const std::wstring delimiter{ L"%HL%" };
+    static const size_t delimiterLength{ delimiter.length() };
+
+    // Find the delimiters.
+    size_t firstSplitPosition = contributeHyperlinkText .find(delimiter, 0);
+    assert(firstSplitPosition != std::wstring::npos);
+    size_t secondSplitPosition = contributeHyperlinkText .find(delimiter, firstSplitPosition + 1);
+    assert(secondSplitPosition != std::wstring::npos);
+    size_t hyperlinkTextLength = secondSplitPosition - (firstSplitPosition + delimiterLength);
+
+    // Assign pieces.
+    auto contributeTextBeforeHyperlink = ref new String(contributeHyperlinkText.substr(0, firstSplitPosition).c_str());
+    auto contributeTextLink = ref new String(contributeHyperlinkText.substr(firstSplitPosition + delimiterLength, hyperlinkTextLength).c_str());
+    auto contributeTextAfterHyperlink = ref new String(contributeHyperlinkText.substr(secondSplitPosition + delimiterLength).c_str());
+
+    ContributeRunBeforeLink->Text = contributeTextBeforeHyperlink;
+    ContributeRunLink->Text = contributeTextLink;
+    ContributeRunAfterLink->Text = contributeTextAfterHyperlink;
+
+}
+
+void SettingsPage::BackButtonClick(Platform::Object ^ sender, Windows::UI::Xaml::RoutedEventArgs ^ e)
+{
+    MainPageProperty->CollapseSettings();
```

Comment:
You should be navigating back to the previous mode, not just collapsing settings.

Reply:
@sanderl If we do that, we'll have to store the previous mode in AppData, then navigate back to it, which is complicated. This just collapses the grid that contains the settings page.